packages/kit/src/core/postbuild/analyse.js

```diff
@@ -93,10 +93,11 @@ async function analyse({ manifest_path, env }) {
 				prerender = mod.prerender;
 			}
 
-			Object.values(mod).forEach((/** @type {import('types').HttpMethod} */ method) => {
+			Object.keys(mod).forEach((key) => {
+				const method = /** @type {import('types').HttpMethod} */ (key);
 				if (mod[method] && ENDPOINT_METHODS.has(method)) {
 					api_methods.push(method);
-				} else if (mod.fallback) {
+				} else if (mod.fallback && key === 'fallback') {
```

Comment:
Actually, I think this would be equivalent and shorter?
```suggestion
				} else if (key === 'fallback') {
```

Reply:
I guess it depends if we want to check the value of `fallback` for truthiness. Is there a case (or is it even possible) where someone conditionally assigns a request handler to it?
e.g., `export const fallback = dev ? requestHandler : undefined;`